gclient/src/node/mod.rs

```diff
@@ -16,4 +16,77 @@
 // You should have received a copy of the GNU General Public License
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
+use result::{Error, Result};
+use std::{
+    ffi::OsStr,
+    io::{BufRead, BufReader},
+    process::{Child, Command, Stdio},
+};
+use ws::WSAddress;
+
+mod port;
+mod result;
 pub mod ws;
+
+/// A struct representing a node running on local PC.
+#[derive(Debug)]
+pub struct Node {
+    process: Child,
+    ws_address: WSAddress,
+}
+
+impl Node {
+    /// Instantiates a node runnin in development mode (--dev) via spawning its
+    /// process using the specified `path` on a randomly picked port. Waits for
+    /// the node getting initialized before returning it to a caller.
+    /// Ideally, the node's binary should be downloaded by means of CI pipeline
+    /// from https://get.gear.rs./
+    pub fn from_path(path: impl AsRef<OsStr>) -> Result<Self> {
```

Comment:
Why not `AsRef<Path>`?

Reply:
Also, I like when methods that return `Result` have a `try_` prefix (similar to those in [`TryFrom` trait](https://doc.rust-lang.org/std/convert/trait.TryFrom.html)).

Reply:
> Why not `AsRef<Path>`?

@shamilsan I have chosen the type based on the one used for the `Command::new` function. Do you think they chose it wrong and the `AsRef<Path>` should've been used instead? I am happy to amend, but please explain why

Reply:
> @shamilsan I have chosen the type based on the one used for the `Command::new` function. Do you think they chose it wrong and the `AsRef<Path>` should've been used instead? I am happy to amend, but please explain why

Ah, I see. It is ok, as both `Path` and `PathBuf` implement `AsRef<OsStr>`. For me, `Path` is just a little bit more habitual.

Reply:
> Also, I like when methods that return `Result` have a `try_` prefix (similar to those in [`TryFrom` trait](https://doc.rust-lang.org/std/convert/trait.TryFrom.html)).

addressed